src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -505,58 +525,72 @@ XdsResolver::XdsConfigSelector::XdsConfigSelector(
         route_action->max_stream_duration =
             resolver_->current_listener_.http_max_stream_duration;
       }
-      Match(
+      absl::StatusOr<std::vector<std::string>> status_or_cluster_names = Match(
           route_action->action,
           // cluster name
           [&](const XdsRouteConfigResource::Route::RouteAction::ClusterName&
-                  cluster_name) {
+                  cluster_name) -> absl::StatusOr<std::vector<std::string>> {
             auto result = CreateMethodConfig(route_entry.route, nullptr);
             if (!result.ok()) {
-              *status = result.status();
-              return;
+              return result.status();
             }
             route_entry.method_config = std::move(*result);
-            MaybeAddCluster(
-                absl::StrCat("cluster:", cluster_name.cluster_name));
+            return std::vector<std::string>{std::string(
```

Comment:
`absl::StrCat()` returns `std::string`, so there's no need to construct a new `std::string` from its result.

Same thing throughout.